haystack/nodes/preprocessor/splitter.py

```diff
@@ -0,0 +1,502 @@
+                        "Set the maximum amout of characters allowed through the 'max_chars' parameter. "
+                        "Keep in mind that very long Documents can severely impact the performance of Readers.",
+                        len(document.content),
+                        split_max_chars,
+                        split_max_chars,
+                        len(document.content) - split_max_chars,
+                    )
+                    hard_splits = [
+                        document.content[pos : pos + split_max_chars]
+                        for pos in range(0, len(document.content), split_max_chars)
+                    ]
+                    sub_units = hard_splits, [0] * len(hard_splits)
+                    sane_documents += self.split_into_units(
+                        document=document,
+                        units=sub_units,
+                        add_page_number=add_page_number,
+                        _start_from_page=document.meta.get("page", 1),
+                    )[0]
+
+            final_documents += sane_documents
+        return {"documents": final_documents}, "output_1"
+
+    def run_batch(  # type: ignore
+        self,
+        documents: List[List[Document]],
+        split_by: Optional[SplitBy] = None,
+        split_regex: Optional[str] = None,
+        split_length: Optional[int] = None,
+        split_overlap: Optional[int] = None,
+        split_max_chars: Optional[int] = None,
+        add_page_number: Optional[bool] = None,
+    ):
+        documents = [
+            self.run(
+                documents=docs,
+                split_by=split_by,
+                split_regex=split_regex,
+                split_length=split_length,
+                split_overlap=split_overlap,
+                split_max_chars=split_max_chars,
+                add_page_number=add_page_number,
+            )[0]["documents"]
+            for docs in tqdm(documents, disable=not self.progress_bar, desc="Splitting", unit="docs")
+        ]
+        return {"documents": documents}, "output_1"
+
+    def split_into_units(
+        self,
+        document: Document,
+        units: Tuple[List[str], List[int]],
+        add_page_number: bool = True,
+        _start_from_page: int = 1,
+    ) -> Tuple[List[Document], List[int]]:
+        """
+        Splits the parent document into single units documents.
+        This algorithm is heavily simplified by the lack of split_length and split_overlap.
+        Those are later applied by the merger.
+        """
+        if isinstance(document, dict):
+            warnings.warn(
+                "Passing a dictionary to Preprocessor.split() is deprecated. Use Document objects.", DeprecationWarning
+            )
+            document = Document.from_dict(document)
+
+        if document.content_type != "text":
+            raise ValueError(
+                f"Document content type is not 'text', but '{document.content_type}'. Preprocessor only handles text documents."
```

Comment:
Make sure the document you pass contains text only as PreProcessor only handles text documents.